museum_export/clean_up_content.py

```diff
@@ -1,37 +1,80 @@
 # clean_up_content.py
 from datetime import datetime
+from add_image_records_as_child_items import AddImageRecordsAsChildItems
 from dependencies.pipelineutilities.creatorField import creatorField  # noqa: E402
+from pipelineutilities.csv_collection import _add_additional_paths
 
 
 class CleanUpContent():
     """ This class accepts a JSON object representing one item of museum content and massages that to fit our needs for processing. """
-    def __init__(self, object):
-        self.cleaned_up_content = CleanUpContent._clean_up_content(object)
+    def __init__(self, config: dict, image_files: dict):
+        self.config = config
+        self.image_files = image_files
+        # self.cleaned_up_content = CleanUpContent._clean_up_content(object, image_files)
 
-    @staticmethod
-    def _clean_up_content(object):
+    def clean_up_content(self, object: dict) -> dict:
         """ This calls all other modules locally """
-        object = CleanUpContent._define_worktype(object)
+        object = self._define_worktype(object)
+        object = self._fix_modified_date(object)
+        object = self._fix_creators(object)
+        object = self._remove_bad_subjects(object)
+        object = self._add_missing_required_fields(object)
+        object = self._clean_up_creation_place(object)
+        object = self._clean_up_object_special_characters(object)
+        if object.get("level", "manifest") != "file" and "digitalAssets" in object:
+            add_image_records_as_child_items_class = AddImageRecordsAsChildItems(self.image_files)
+            object = add_image_records_as_child_items_class.add_images_as_children(object)
+        if "children" in object and len(object["children"]) == 0:
+            del object["children"]
+        _add_additional_paths(object, self.config)
+        if "items" in object:
+            for item in object["items"]:
+                self.clean_up_content(item)
+        return object
+
+    def _clean_up_object_special_characters(self, object: dict) -> dict:
+        field_names_to_clean = ["title", "copyrightStatement", "description"]
+        for name in field_names_to_clean:
+            if name in object:
+                object[name] = self._replace_special_characters(object[name])
+        return object
+
+    def _clean_up_creation_place(self, object: dict) -> dict:
+        if "creationPlace" in object:
+            if "county" in object["creationPlace"]:
+                object["creationPlace"]["county"] = self._replace_special_characters(object["creationPlace"]["county"])
+        return object
+
+    def _replace_special_characters(self, field_string: str) -> str:
```

Comment:
Do we want to try a general conversion here rather than a specific one?  Just because I imagine there are more than this list?  
As with this: https://stackoverflow.com/questions/663058/html-entity-codes-to-text

Reply:
I added code in _replace_sepcial_characters to make a call as you suggested to handle html escaped sequences.  I also had to keep one line to translate %20 to a space.